javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,169 @@
+"use strict"
+/**Voice assistant */
+/**variable name save the name and use for respond to commond */
+let name = '';
+/** keep track of todos */
+const todos = [];
+
+/**main function of the program which return a response that corresponds to the command */
+function getReply(command) {
+    //check the commond is not a string then stop the program with a message 
+    if (typeof command !== 'string') {
+        return 'Please give command in correct format';
+    }
+    //convert command into an array
+    const commandArray = command.split(' ');
+    //if the first word of comman is hello then reply with wellcome message
+    if (commandArray[0].toLowerCase() === 'hello') {
+        if (name === '') {
+            name = commandArray[commandArray.length - 1]
```

Comment:
What you did is correct. There is also an alternative like below 
There are string methods like 'startsWith' and 'endsWith'.
For example 
 const commandWords = inputString.split(' ');

  if (inputString.startsWith('hello my name is')) {
    name = commandWords[commandWords.length - 1];
    return `Nice to meet you ${name}`;
  }

Reply:
Thanks you mentioning this , now I have implement it